lib/fizzy/parser.cpp

```diff
@@ -642,22 +642,23 @@ Module parse(bytes_view input)
             throw validation_error{"duplicate export name " + export_.name};
     }
 
-    if (module.startfunc)
+    if (module->startfunc)
     {
-        if (*module.startfunc >= total_func_count)
+        if (*module->startfunc >= total_func_count)
             throw validation_error{"invalid start function index"};
 
-        const auto& func_type = module.get_function_type(*module.startfunc);
+        const auto& func_type = module->get_function_type(*module->startfunc);
         if (!func_type.inputs.empty() || !func_type.outputs.empty())
             throw validation_error{"invalid start function type"};
     }
 
     // Process code. TODO: This can be done lazily.
-    module.codesec.reserve(code_binaries.size());
+    module->codesec.reserve(code_binaries.size());
     for (size_t i = 0; i < code_binaries.size(); ++i)
-        module.codesec.emplace_back(parse_code(code_binaries[i], static_cast<FuncIdx>(i), module));
+        module->codesec.emplace_back(
+            parse_code(code_binaries[i], static_cast<FuncIdx>(i), *module));
 
-    return module;
+    return std::unique_ptr<const Module>(module.release());
```

Comment:
This is not needed. See (6) in https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr.

Reply:
I know, it was not there first, but then some warning/sanitizers comlained.

Specifically, one warning claimed that returning different, but compatible type may lead to additional copying on some older compilers, and `std::move(module)` was adviced to force the move.

And then another one warned against return with `std::move` :-)

Reply:
This is the fist one https://app.circleci.com/pipelines/github/wasmx/fizzy/5103/workflows/6b7d676a-32e8-4e37-98f0-1de319f51f46/jobs/38251

And the second one https://app.circleci.com/pipelines/github/wasmx/fizzy/5104/workflows/a0c3b9ce-c732-49d4-8f30-e578d3f5b764/jobs/38266

Reply:
Please disable [`-Wreturn-std-move-in-c++11`](https://clang.llvm.org/docs/DiagnosticsReference.html#wreturn-std-move-in-c-11) in Clang.